pkg/storage/couchdb/couchdbstore.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package couchdbstore
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	_ "github.com/go-kivik/couchdb" // The CouchDB driver
+	"github.com/go-kivik/kivik"
+
+	"github.com/trustbloc/edge-core/pkg/storage"
+)
+
+// Provider represents an CouchDB implementation of the storage.Provider interface
+type Provider struct {
+	hostURL       string
+	couchDBClient *kivik.Client
+}
+
+const blankHostErrMsg = "hostURL for new CouchDB provider can't be blank"
+
+// NewProvider instantiates Provider
+func NewProvider(hostURL string) (*Provider, error) {
+	if hostURL == "" {
+		return nil, errors.New(blankHostErrMsg)
+	}
+
+	client, err := kivik.New("couch", hostURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Provider{hostURL: hostURL, couchDBClient: client}, nil
+}
+
+// OpenStore opens and returns a store for the given name.
+func (p *Provider) OpenStore(name string) (storage.Store, error) {
+	exists, err := p.couchDBClient.DBExists(context.Background(), name)
```

Comment:
why are we still doing this? This stuff gave us lots of headaches in the past. The solution back then was to have a "leader" create the DB if it doesn't exist.

Why isn't this a devops issue?